### PR TITLE
#166265830 Integrate postgreSQL as the project database

### PR DESCRIPTION
**Pull request by chuxmykel**
#### What does this PR do?
This PR integrates PostgreSQL as the project database using sequelize as an ORM. It also provides scripts for running basic database operations such as migrations and seeding.

#### Description of Task to be completed?
- install sequelize as ORM and sequelize-cli for operations
- add users migrations and models for table creation and DB interaction
- update .env.sample file to contain database URL
- add database related scripts
- update README.md file to describe new database scripts

#### How should this be manually tested?
- clone the REPO cd into the directory and create a `.env` file from the `.env.sample` file provided
- run `npm install`
- follow the migrations guide on the README.md file to run different operations on the DB

#### Any background context you want to provide?
- None

#### What are the relevant pivotal tracker stories?
[#166265830](https://www.pivotaltracker.com/story/show/166265830)

#### Screenshots (if appropriate)
Screenshot from 2019-05-29 12-26-13
Screenshot from 2019-05-29 12-25-03
Screenshot from 2019-05-29 12-24-11
Screenshot from 2019-05-29 12-24-05

### CODE REVIEW

File: server/app/models/index.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+'use strict';
```
Comment (cvjude) [inline server/app/models/index.js:1]: I don't think this Line is necessary
Node is in strict mode

File: server/app/models/user.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+'use strict';
```
Comment (cvjude) [inline server/app/models/user.js:1]: Here too

File: server/app/seeders/20190528204615-demo-user.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+'use strict';
```
Comment (cvjude) [inline server/app/seeders/20190528204615-demo-user.js:1]: Here too

**Review by encodedBicoding — APPROVED** [review]

File: server/app/models/index.js (hunk 1 above)
Comment (chuxmykel) [reply server/app/models/index.js:1]: This file was automatically generated by sequelize-cli. Would you suggest I remove it now or maybe later in a branch for refactoring.

File: server/app/models/user.js (hunk 2 above)
Comment (chuxmykel) [reply server/app/models/user.js:1]: [same reply](https://github.com/emp-daisy/HelloBooksLib/pull/11#discussion_r288521602)

File: server/app/seeders/20190528204615-demo-user.js (hunk 3 above)
Comment (chuxmykel) [reply server/app/seeders/20190528204615-demo-user.js:1]: [same reply](https://github.com/emp-daisy/HelloBooksLib/pull/11#discussion_r288521602)

File: server/app/models/index.js (hunk 1 above)
Comment (cvjude) [reply server/app/models/index.js:1]: Ok boss

**Review by cvjude — APPROVED** [review]

**Review by OvieMudi — COMMENTED** [review]
Why is Hound CI nagging when you used built-in Node variables/functions like ` __dirname, require` etc?

**Comment by chuxmykel** [comment]
@OvieMudi I have no idea. I may have to look it up

**Comment by chuxmykel** [comment]
@OvieMudi [check this out](http://help.houndci.com/configuration/eslint)
We may need to do some config in our .eslintrc file

**Comment by Hadeneekeh** [comment]
@chuxmykel that link is referring to the eslint configuration to add to the hound.yml file which @PelumiAlesh have taken care of.

**Pull request merged**